flutter_secure_storage/android/src/main/java/com/it_nomads/fluttersecurestorage/FlutterSecureStorage.java

```diff
@@ -102,18 +96,21 @@ private void ensureOptions() {
                 : DEFAULT_KEY_PREFIX;
     }
 
-    private SharedPreferences getEncryptedSharedPreferences() {
-        if (encryptedPreferences == null) {
+    private SharedPreferences getEncryptedSharedPreferences() throws GeneralSecurityException, IOException {
+        try {
+            final SharedPreferences encryptedPreferences = initializeEncryptedSharedPreferencesManager(applicationContext);
+            migrateToEncryptedPreferences(encryptedPreferences);
+            return encryptedPreferences;
+        } catch (Exception e) {
```

Comment:
I am a bit unsure, if we should reset the storage on *any* exception; maybe, we should only reset it on exceptions where we know that the reset is absolutely necessary? (I think, `javax.crypto.AEADBadTagException` would be an example)